# **Exploratory Data Analysis**

### **Q1. What is the average, median, and standard deviation of house prices?**

In [ ]:
from google.colab import files
uploaded = files.upload()

Saving House Price India.csv to House Price India.csv


In [ ]:
import pandas as pd

df = pd.read_csv("House Price India.csv")

# Calculate statistics for the Price column
average_price = df["Price"].mean()
median_price = df["Price"].median()
std_price = df["Price"].std()

print("Average House Price:", average_price)
print("Median House Price:", median_price)
print("Standard Deviation of House Price:", std_price)

Average House Price: 538932.2183310534
Median House Price: 450000.0
Standard Deviation of House Price: 367532.3808039669


Interpretation

Average (Mean) Price = 538,932.22

*  This represents the average cost of houses in the dataset.
*  On average, a house costs approximately 538,932.

Median Price = 450,000

*  Half of the houses have prices below 450,000 and half above.
*  Since the median is lower than the mean, the dataset likely contains some high-priced houses that increase the average.

Standard Deviation = 367,532.38

*  This measures how much house prices vary from the average.
*  A large standard deviation indicates that house prices are widely spread out, ranging from relatively inexpensive to very expensive properties.

Conclusion

The housing market in this dataset shows considerable price variation. The mean being higher than the median suggests a right-skewed distribution, meaning a small number of expensive houses are pulling the average upward.





### **Q2. Which number of bedrooms is most common?**

In [ ]:
import pandas as pd

# Load the dataset
df = pd.read_csv("House Price India.csv")

# Find the most common number of bedrooms
most_common_bedrooms = df["number of bedrooms"].mode()[0]

print("Most Common Number of Bedrooms:", most_common_bedrooms)

# Frequency of each bedroom count
bedroom_counts = df["number of bedrooms"].value_counts()
print(bedroom_counts)

Most Common Number of Bedrooms: 3
number of bedrooms
3     6612
4     4724
2     1844
5     1079
6      176
1      136
7       30
8       11
9        3
10       3
33       1
11       1
Name: count, dtype: int64


Interpretation

*  The mode represents the bedroom count that appears most frequently in the dataset.
*  This indicates the type of house that is most commonly available in the housing market.

*  The frequency table helps compare the popularity of different bedroom configurations.





### **Q3. Check if the price data is skewed.**

In [5]:
from google.colab import files
uploaded = files.upload()

Saving House Price India.csv to House Price India.csv


In [6]:
import pandas as pd

# Load the dataset
df = pd.read_csv("House Price India.csv")

In [7]:
print(df.columns)

Index(['id', 'Date', 'number of bedrooms', 'number of bathrooms',
       'living area', 'lot area', 'number of floors', 'waterfront present',
       'number of views', 'condition of the house', 'grade of the house',
       'Area of the house(excluding basement)', 'Area of the basement',
       'Built Year', 'Renovation Year', 'Postal Code', 'Lattitude',
       'Longitude', 'living_area_renov', 'lot_area_renov',
       'Number of schools nearby', 'Distance from the airport', 'Price'],
      dtype='object')


In [8]:
skewness = df['Price'].skew()

print("Skewness:", skewness)

if skewness > 0:
    print("The Price data is positively (right) skewed.")
elif skewness < 0:
    print("The Price data is negatively (left) skewed.")
else:
    print("The Price data is approximately symmetric.")

Skewness: 4.269297720707116
The Price data is positively (right) skewed.


Interpretation



*   The skewness value is 4.2693, which is much greater than 0
*   The mean price (538,932) is higher than the median price (450,000).
*   This indicates that the distribution of house prices is strongly positively (right) skewed.
*   A small number of very expensive houses are pulling the average price upward.





### **Q4. What is the average price for each number of bedrooms?**

In [10]:
from google.colab import files
uploaded = files.upload()

Saving House Price India.csv to House Price India (1).csv


In [11]:
import pandas as pd

# Load the dataset
df = pd.read_csv("House Price India.csv")

In [12]:
# Check column names first
print(df.columns)

# Average price by number of bedrooms
avg_price_by_bedrooms = df.groupby('number of bedrooms')['Price'].mean()

print(avg_price_by_bedrooms)

Index(['id', 'Date', 'number of bedrooms', 'number of bathrooms',
       'living area', 'lot area', 'number of floors', 'waterfront present',
       'number of views', 'condition of the house', 'grade of the house',
       'Area of the house(excluding basement)', 'Area of the basement',
       'Built Year', 'Renovation Year', 'Postal Code', 'Lattitude',
       'Longitude', 'living_area_renov', 'lot_area_renov',
       'Number of schools nearby', 'Distance from the airport', 'Price'],
      dtype='object')
number of bedrooms
1     3.089638e+05
2     3.985476e+05
3     4.632776e+05
4     6.361988e+05
5     7.752550e+05
6     8.375815e+05
7     1.016544e+06
8     1.208455e+06
9     7.766663e+05
10    8.200000e+05
11    5.200000e+05
33    6.400000e+05
Name: Price, dtype: float64


Interpretation



*   The result shows the average house price for each bedroom count.
*   Compare the averages to see how house prices change as the number of  bedrooms increases.
*  Generally, houses with more bedrooms tend to have higher average prices, although location and other factors can also influence price.






### **Q5.  What is the relationship between living area and price? (Use : Area without Basement).**

In [13]:
import pandas as pd

# Load the dataset
df = pd.read_csv("House Price India.csv")

In [14]:
# Check column names first
print(df.columns)

# Correlation between Area without Basement and Price
correlation = df['Area of the house(excluding basement)'].corr(df['Price'])

print("Correlation:", correlation)

Index(['id', 'Date', 'number of bedrooms', 'number of bathrooms',
       'living area', 'lot area', 'number of floors', 'waterfront present',
       'number of views', 'condition of the house', 'grade of the house',
       'Area of the house(excluding basement)', 'Area of the basement',
       'Built Year', 'Renovation Year', 'Postal Code', 'Lattitude',
       'Longitude', 'living_area_renov', 'lot_area_renov',
       'Number of schools nearby', 'Distance from the airport', 'Price'],
      dtype='object')
Correlation: 0.6152204198678108


Interpretation



*   Correlation close to +1 → Strong positive relationship
*   Correlation between 0.5 and 0.8 → Moderate positive relationship


*   Correlation close to 0 → Weak or no relationship
*   Correlation close to -1 → Strong negative relationship





### **Q6. Identify any anomalies where houses have high prices but low areas.**

In [16]:
import pandas as pd

# Load the dataset
df = pd.read_csv("House Price India.csv")

In [17]:
# Calculate average price and average area
avg_price = df['Price'].mean()
avg_area = df['Area of the house(excluding basement)'].mean()

# Find anomalies: High Price but Low Area
anomalies = df[
    (df['Price'] > avg_price) &
    (df['Area of the house(excluding basement)'] < avg_area)
]

print("Number of anomalies:", len(anomalies))
print(anomalies[['Price', 'Area of the house(excluding basement)']])

Number of anomalies: 1887
        Price  Area of the house(excluding basement)
5      790000                                   1700
7      750000                                   1550
8      750000                                   1440
9      698000                                   1300
11     650000                                   1640
...       ...                                    ...
14576  931000                                   1560
14578  875000                                   1160
14589  600000                                   1000
14591  579000                                   1800
14593  565000                                   1240

[1887 rows x 2 columns]


Interpretation



*   These houses are priced above the average house price.
*   Their living area is below the average living area.
*   Such properties may be expensive due to factors other than size, such as:


*   Prime location
*   Luxury features
*   Recent renovations
*   Better neighborhood amenities
*   Waterfront or premium views













### **Q7. Compare average price based on number of floors and houses with or without waterfront.**

In [19]:
import pandas as pd

# Load the dataset
df = pd.read_csv("House Price India.csv")

In [20]:
avg_price_by_floors = df.groupby('number of floors')['Price'].mean()

print("Average Price by number of floors:")
print(avg_price_by_floors)

Average Price by number of floors:
number of floors
1.0    4.369769e+05
1.5    5.549301e+05
2.0    6.487372e+05
2.5    1.106233e+06
3.0    5.663281e+05
3.5    1.205875e+06
Name: Price, dtype: float64


In [21]:
avg_price_by_waterfront = df.groupby('waterfront present')['Price'].mean()

print("\nAverage Price by waterfront present:")
print(avg_price_by_waterfront)


Average Price by waterfront present:
waterfront present
0    5.304174e+05
1    1.641902e+06
Name: Price, dtype: float64


In [22]:
comparison = df.groupby(
    ['number of floors', 'waterfront present']
)['Price'].mean().reset_index()

print(comparison)

    number of floors  waterfront present         Price
0                1.0                   0  4.323724e+05
1                1.0                   1  1.211081e+06
2                1.5                   0  5.476162e+05
3                1.5                   1  1.232507e+06
4                2.0                   0  6.357742e+05
5                2.0                   1  2.075931e+06
6                2.5                   0  1.087654e+06
7                2.5                   1  3.280000e+06
8                3.0                   0  5.558095e+05
9                3.0                   1  1.655000e+06
10               3.5                   0  1.205875e+06


Interpretation

The average house price was compared across different numbers of floors and waterfront status. Houses with more floors generally tend to have higher average prices due to increased living space. Houses with waterfront access typically have significantly higher average prices than those without waterfront access, indicating that location and scenic value strongly influence property prices.

If the maximum price is much larger than the minimum price, it indicates that the dataset contains a wide range of properties, from affordable houses to luxury homes. This large gap often contributes to a positively skewed price distribution, where a few very expensive houses increase the overall average price.

### **Q8. Identify the minimum and maximum house price. What does this indicate?**

In [23]:
import pandas as pd

# Load the dataset
df = pd.read_csv("House Price India.csv")

In [24]:
# Find minimum and maximum house price
min_price = df['Price'].min()
max_price = df['Price'].max()

print("Minimum House Price:", min_price)
print("Maximum House Price:", max_price)

Minimum House Price: 78000
Maximum House Price: 7700000


Interpretation

*   Minimum House Price represents the least expensive property in the dataset.
*   Maximum House Price represents the most expensive property in the dataset.
*   The difference between these values indicates the price range of houses.





### **Q9. Which location (zipcode/area) has the highest average price?**

In [25]:
import pandas as pd

# Load the dataset
df = pd.read_csv("House Price India.csv")

In [27]:
# Average price by zipcode
avg_price_zipcode = df.groupby('Postal Code')['Price'].mean()

# Zipcode with highest average price
highest_price_zipcode = avg_price_zipcode.idxmax()
highest_avg_price = avg_price_zipcode.max()

print("Postal Code with Highest Average Price:", highest_price_zipcode)
print("Highest Average Price:", highest_avg_price)

Postal Code with Highest Average Price: 122071
Highest Average Price: 2348310.810810811


Interpretation

The zipcode/area with the highest average price represents the most expensive location in the dataset. Houses in this area are likely valued higher due to factors such as premium location, better amenities, accessibility, neighborhood quality, or demand.

### **Q10. Write at least 5 insights from your analysis.**


*  House prices are positively skewed

    The mean house price is higher than the median price, indicating that a small number of very expensive houses increase the overall average.
*   Larger houses tend to be more expensive

    There is a positive relationship between Area without Basement and Price, showing that houses with larger living areas generally have higher market values.

*   The number of bedrooms influences house prices

    Houses with more bedrooms typically have higher average prices because they offer more living space and utility.
*   Waterfront properties are more valuable

    Houses with waterfront access have a higher average price compared to non-waterfront properties, suggesting that location and scenic views significantly affect property value.

*   House prices vary considerably across locations

    Some zipcodes/areas have much higher average prices than others, indicating that location is one of the strongest factors influencing house prices.